pkg/sdk/validation/typeinstance.go

```diff
@@ -0,0 +1,143 @@
+	var typeInstanceCollection []*typeInstanceData
+	typeRefCollection := TypeRefCollection{}
+
+	for _, ti := range typeInstance.TypeInstances {
+		if ti == nil {
+			continue
+		}
+		typeRef := types.TypeRef{
+			Path:     ti.TypeRef.Path,
+			Revision: ti.TypeRef.Revision,
+		}
+		name := getManifestPathWithRevision(ti.TypeRef.Path, ti.TypeRef.Revision)
+		typeRefCollection[name] = TypeRef{
+			TypeRef: typeRef,
+		}
+		typeInstanceCollection = append(typeInstanceCollection, &typeInstanceData{
+			alias:               ti.Alias,
+			value:               ti.Value,
+			typeRefWithRevision: name,
+		})
+	}
+
+	schemasCollection, err := ResolveTypeRefsToJSONSchemas(ctx, client, typeRefCollection)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while resolving TypeRefs to JSON Schemas")
+	}
+	return validateTypeInstances(schemasCollection, typeInstanceCollection)
+}
+
+// ValidateTypeInstanceToUpdate  is responsible for validating TypeInstance which exists and will be updated.
+func ValidateTypeInstanceToUpdate(ctx context.Context, client TypeInstanceValidationHubClient, typeInstanceToUpdate []graphqllocal.UpdateTypeInstancesInput) (Result, error) {
+	var typeInstanceIds []string
+	idToTypeNameMap := map[string]string{}
+	for _, ti := range typeInstanceToUpdate {
+		typeInstanceIds = append(typeInstanceIds, ti.ID)
+	}
+
+	typeInstancesTypeRef, err := client.FindTypeInstancesTypeRef(ctx, typeInstanceIds)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while finding TypeInstance Type reference")
+	}
+
+	typeRefCollection := TypeRefCollection{}
+	for id, typeReference := range typeInstancesTypeRef {
+		name := getManifestPathWithRevision(typeReference.Path, typeReference.Revision)
+		typeRefCollection[name] = TypeRef{
+			TypeRef: types.TypeRef{
+				Path:     typeReference.Path,
+				Revision: typeReference.Revision,
+			},
+		}
+		idToTypeNameMap[id] = name
+	}
+
+	var typeInstanceCollection []*typeInstanceData
+	for _, ti := range typeInstanceToUpdate {
+		if ti.TypeInstance == nil {
+			continue
+		}
+		typeInstanceCollection = append(typeInstanceCollection, &typeInstanceData{
+			id:                  ti.ID,
+			value:               ti.TypeInstance.Value,
+			typeRefWithRevision: idToTypeNameMap[ti.ID],
+		})
+	}
+
+	schemasCollection, err := ResolveTypeRefsToJSONSchemas(ctx, client, typeRefCollection)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while resolving TypeRefs to JSON Schemas")
+	}
+
+	return validateTypeInstances(schemasCollection, typeInstanceCollection)
+}
+
+func validateTypeInstances(schemaCollection SchemaCollection, typeInstanceCollection []*typeInstanceData) (Result, error) {
+	resultBldr := NewResultBuilder("Validation TypeInstances")
+
+	for _, ti := range typeInstanceCollection {
+		if _, ok := ti.value.(map[string]interface{}); !ok {
+			return Result{}, errors.New("could not create map from TypeInstance Value")
+		}
```

Comment:
This statement should cover the case when the user tries to edit TypeInstance and delete the values from it. Then, it would not be converted to `map(string)interface` and validation would not pass.